snapcraft/tests/pluginhandler/test_pluginhandler.py

```diff
@@ -1821,6 +1821,54 @@ def test_clean_part_remaining_parts(self, mock_exists, mock_listdir,
         mock_listdir.assert_called_once_with(partdir)
         self.assertFalse(mock_rmdir.called)
 
+    def test_clean_dir_symlinks(self):
+        # Create part1 and get it through the "build" step.
+        handler1 = mocks.loadplugin('part1')
+        handler1.makedirs()
+
+        bindir = os.path.join(handler1.code.installdir, 'bin')
+        os.makedirs(bindir)
+        path = os.path.join(bindir, '1')
+        os.mkdir(path)
+        symlink_path = os.path.join(bindir, '1-link')
+        os.symlink(path, symlink_path)
+
+        handler1.mark_done('build')
+        handler1.stage()
+
+        stage_symlink_path = os.path.join(handler1.stagedir, 'bin', '1-link')
+        stage_path = os.path.join(handler1.stagedir, 'bin', '1')
+        self.assertTrue(os.path.exists(stage_symlink_path))
+        self.assertTrue(os.path.exists(stage_path))
```

Comment:
The assertion errors are way better if you kneel to testtools: `self.assertThat(stage_path, FileExists())`. This way you can even use `self.assertThat(stage_symlink_path, tests.LinkExists('1'))` that will verify that the link exists and is pointing to '1'.